zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/AuthorizationArchTest.java

```diff
@@ -0,0 +1,183 @@
+
+  private static DescribedPredicate<JavaClass> thatProcessCommands() {
+    return new DescribedPredicate<>("process commands") {
+      @Override
+      public boolean test(final JavaClass javaClass) {
+        return Predicates.implement(TypedRecordProcessor.class)
+            // Not all processors use the TypedRecordProcessor interface. We also need to check
+            // the
+            // CommandProcessor and the UserTaskCommandProcessor interfaces.
+            .or(Predicates.implement(CommandProcessor.class))
+            .or(Predicates.implement(UserTaskCommandProcessor.class))
+            .test(javaClass);
+      }
+    };
+  }
+
+  private static DescribedPredicate<JavaClass> thatAreNotExcludedFromAuthorizationChecks() {
+    return new DescribedPredicate<>("are not excluded from authorization checks") {
+      @Override
+      public boolean test(final JavaClass javaClass) {
+        // Not all processors need to check authorization. We need to exclude those processors.
+        return !getProcessorsThatDoNotCheckAuthorization().contains(javaClass.reflect());
+      }
+    };
+  }
+
+  private static List<Class<?>> getProcessorsThatDoNotCheckAuthorization() {
+    final List<Class<?>> processors = new ArrayList<>();
+    processors.add(BpmnStreamProcessor.class);
+    processors.add(CommandProcessorImpl.class);
+    processors.add(DeploymentDistributeProcessor.class);
+    processors.add(DeploymentDistributionCompleteProcessor.class);
+    processors.add(CommandDistributionAcknowledgeProcessor.class);
+    processors.add(CommandDistributionContinueProcessor.class);
+    processors.add(CommandDistributionFinishProcessor.class);
+    processors.add(JobBatchActivateProcessor.class); // TODO REMOVE THIS ONE
+    processors.add(JobCancelProcessor.class);
+    processors.add(JobRecurProcessor.class);
+    processors.add(JobTimeOutProcessor.class);
+    processors.add(JobYieldProcessor.class);
+    processors.add(MessageBatchExpireProcessor.class);
+    processors.add(MessageExpireProcessor.class);
+    processors.add(MessageSubscriptionCorrelateProcessor.class);
+    processors.add(MessageSubscriptionDeleteProcessor.class);
+    processors.add(MessageSubscriptionMigrateProcessor.class);
+    processors.add(MessageSubscriptionCreateProcessor.class);
+    processors.add(MessageSubscriptionRejectProcessor.class);
+    processors.add(ProcessMessageSubscriptionCorrelateProcessor.class);
+    processors.add(ProcessMessageSubscriptionCreateProcessor.class);
+    processors.add(ProcessMessageSubscriptionDeleteProcessor.class);
+    processors.add(ProcessInstanceBatchActivateProcessor.class);
+    processors.add(ProcessInstanceBatchTerminateProcessor.class);
+    processors.add(ProcessInstanceCreationCreateWithResultProcessor.class);
+    processors.add(TimerCancelProcessor.class);
+    processors.add(TimerTriggerProcessor.class);
+    processors.add(UserTaskProcessor.class);
+    return processors;
+  }
```

Comment:
🔧 Did you consider creating an annotation interface like `@ExcludeAuthorizationCheck` that we can use to annotate processors that shouldn't perform this check?

This will be more readable to devs, and we won't need to update this class whenever we add a new processor. We will just need to add the following to this test:

```
.and()
.areNotAnnotatedWith(ExcludeAuthorizationCheck.class)
```